pkg/githubreconciler/client.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2025 Chainguard, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package githubreconciler
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/chainguard-dev/clog"
+	"github.com/google/go-github/v72/github"
+	"golang.org/x/oauth2"
+)
+
+// TokenSourceFunc is a function that creates an OAuth2 token source for a given org/repo.
```

Comment:
Is there a reason we can't use this instead of creating a new one, or update that one?
https://github.com/chainguard-dev/terraform-infra-common/blob/c13ca4656a52fc90160e8134c2d4cabf234173cb/modules/github-bots/sdk/github.go#L36

The sprawl of various GH clients seems

Reply:
Yeah, I originally built this in mono and moved it over, so this can be combined.

Generally the stuff in `modules/` is specific to the binaries in that module, so it probably makes sense to move this stuff into `pkg/`.

I'll get this cleaned up when I have a break

Reply:
done, in a second commit to keep things a bit cleaner